metadata-ingestion/src/datahub/ingestion/source/aws_common.py

```diff
@@ -0,0 +1,79 @@
+from functools import reduce
+from typing import List, Optional, Union
+
+import boto3
+
+from datahub.configuration import ConfigModel
+from datahub.configuration.common import AllowDenyPattern
+
+
+def assume_role(
+    role_arn: str, aws_region: str, credentials: Optional[dict] = None
+) -> dict:
+    credentials = credentials or {}
+    sts_client = boto3.client(
+        "sts",
+        region_name=aws_region,
+        aws_access_key_id=credentials.get("AccessKeyId"),
+        aws_secret_access_key=credentials.get("SecretAccessKey"),
+        aws_session_token=credentials.get("SessionToken"),
+    )
+
+    assumed_role_object = sts_client.assume_role(
+        RoleArn=role_arn, RoleSessionName="DatahubIngestionSource"
+    )
+    return assumed_role_object["Credentials"]
+
+
+class AwsSourceConfig(ConfigModel):
```

Comment:
This is awesome!! 
nit: Should we have a directory for all AWS ingest that shares this config?

Reply:
Good idea – I'll add it as a comment for now